webmap/src/control/ContextMenuControl.ts

```diff
@@ -20,42 +21,63 @@ export default class ContextMenuControl extends L.Control {
     private _init(): void {
         this._pl3xmap.map.on('contextmenu', this._show, this);
         this._pl3xmap.map.on('click', this._hide, this);
-        if (this._customHtml.enabled) {
-            const style = L.DomUtil.create('style', 'leaflet-control-contextmenu-custom-style', document.head);
-            style.innerHTML = this._customHtml.css;
+
+        const css = this._pl3xmap.worldManager.currentWorld?.settings.ui.contextMenu.css;
+        if (css !== undefined) {
+            insertCss(css, this._id);
         }
     }
 
     onAdd(): HTMLDivElement {
         this._dom = L.DomUtil.create('div', 'leaflet-control leaflet-control-contextmenu');
         this._dom.dataset.label = this._pl3xmap.settings!.lang.contextMenu.label;
-        if (this._customHtml.enabled) {
-            this._dom.innerHTML = this._customHtml.html;
-        }
         return this._dom;
     }
 
+
+    onRemove(): void {
+        removeCss(this._id);
+    }
+
     private _show(event: L.LeafletMouseEvent): void {
-        L.DomEvent.stopPropagation(event);
-        if (!this._customHtml.enabled) {
-            this._dom.innerHTML = '';
-            this._getItems(event).forEach((item) => {
-                const menuItem = L.DomUtil.create('div', 'leaflet-control-contextmenu-item', this._dom);
-                menuItem.innerHTML = item.label;
-                L.DomEvent.on(menuItem, 'click', (e) => {
-                    L.DomEvent.stopPropagation(e);
-                    item.callback();
-                    this._hide();
-                });
-            });
+        // Ignore right-clicks on controls (https://github.com/JLyne/LiveAtlas/blob/0819cdf2728b49d361f9adfda09ff08311a59337/src/components/map/MapContextMenu.vue#L188-L194)
+        if(event.originalEvent.target && (event.originalEvent.target as HTMLElement).closest('.leaflet-control')) {
```

Comment:
The right-clicks on controls were ignored even without this code. At least I think so?